main/gradebook/lib/be/category.class.php

```diff
@@ -947,7 +947,11 @@ public function calc_score(
                 /** @var EvalLink|ExerciseLink $link */
                 foreach ($links as $link) {
                     $link->setStudentList($this->getStudentList());
-                    $linkres = $link->calc_score($stud_id, null);
+	                
+	                //Dont forget to add session id
+	                if($session_id) $link->set_session_id($session_id);
```

Comment:
It should be (format code). Also for spacing, we use 4 space instead of tabs.

```
if ($session_id) {
    $link->set_session_id($session_id);
}
```